doc/_extensions/zephyr/doxyrunner.py

```diff
@@ -169,16 +185,16 @@ def doxygen_input_has_changed(doxyfile: str, cache_dir: Path) -> bool:
     if not file_patterns:
         raise ValueError("No FILE_PATTERNS set in Doxyfile")
 
-    # build a dict of input files <-> current modification time
-    files = dict()
+    # build a set with input files hash
+    files = set()
     for file in input_files:
         path = Path(file)
         if path.is_file():
-            files[path.as_posix()] = path.stat().st_mtime_ns
+            files.add(hash_file(path))
```

Comment:
What is the performance impact?

Reply:
I've just done a quick measurement:

Previous method: 0.032463788986206055s
New method: 0.07107734680175781 s

Reply:
2x slowdown!? We can't possibly accept that :stuck_out_tongue_winking_eye:

Just kidding. Thanks for measuring.